core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -2093,6 +2093,9 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
       Seq("hostC", "hostB", "hostD").sorted)
```

Comment:
Add a driver block manager to this test ? It will catch the issue being fixed in `BlockManagerMasterEndpoint`, right ?

Reply:
I tried adding test but like I mentioned in this [comment](https://github.com/apache/spark/pull/33425#discussion_r673340152) currently `BlockManagerMaster` is mocked due to which it doesn't registered and won't be in `blockManagerIdByExecutor`. That is why I didn't add the test.

Reply:
Why would this not work ?

```
    assert(master.getExecutorEndpointRef(SparkContext.DRIVER_IDENTIFIER).isEmpty)
    makeBlockManager(100, SparkContext.DRIVER_IDENTIFIER,
      transferService = Some(new MockBlockTransferService(10, "host0")))
    assert(master.getExecutorEndpointRef(SparkContext.DRIVER_IDENTIFIER).isDefined)
```